pandas/io/parsers.py

```diff
@@ -110,8 +114,9 @@
     are duplicate names in the columns.
 dtype : Type name or dict of column -> type, default None
     Data type for data or columns. E.g. {'a': np.float64, 'b': np.int32}
-    (Unsupported with engine='python'). Use `str` or `object` to preserve and
-    not interpret dtype.
+    Use `str` or `object` to preserve and not interpret dtype.
+    If converters are specified, they will be applied AFTER
+    dtype conversion.
```

Comment:
I was wrong, the c-parser actually ignores the dtype and just uses the converter - changed docstring and added matching test.